test/jasmine/tests/bar_test.js

```diff
@@ -1328,7 +1328,51 @@ describe('A bar plot', function() {
         .then(function() {
             _assertNumberOfBarTextNodes(3);
         })
-        .catch(fail)
+        .catch(failTest)
+        .then(done);
+    });
+
+    it('should update axis range according to visible edits', function(done) {
+        var schema = Plotly.PlotSchema.get();
+        expect(schema.traces.bar.attributes.visible.editType)
+            .toBe('plot', 'visible editType');
+
+        function _assert(msg, xrng, yrng, calls) {
+            var fullLayout = gd._fullLayout;
+            expect(fullLayout.xaxis.range).toBeCloseToArray(xrng, 2, msg + ' xrng');
+            expect(fullLayout.yaxis.range).toBeCloseToArray(yrng, 2, msg + ' yrng');
+
+            var setPositions = gd._fullData[0]._module.setPositions;
+            expect(setPositions).toHaveBeenCalledTimes(calls);
+            setPositions.calls.reset();
+        }
+
+        Plotly.plot(gd, [
+            {type: 'bar', x: [1, 2, 3], y: [1, 2, 1]},
+            {type: 'bar', x: [1, 2, 3], y: [-1, -2, -1]}
+        ])
```

Comment:
Is there a test with stacked bars, where the extremes of one trace depend on the traces before it (so may need to be recalculated if an earlier trace changes visibility)?

Reply:
> Is there a test with stacked bars,

Thanks for writing this down. Visibility restyles are broken on stacked bars, but for a different reason: `Bar.setPositions` mutates `calcdata[i][j].b` which is first set in `Bar.calc` :upside_down_face: 

> where the extremes of one trace depend on the traces before it 

Yep, that's exactly why commit https://github.com/plotly/plotly.js/commit/4c250c3c90f061ecbe1836f8ef07ed9b561d9f6b is very important, 'plot' edit types now go call `setPositions` for this reason.